### ISSUE TITLE
Delay is not applied

### ISSUE THREAD

**Issue by djegithub** [shutter-project/shutter#560]
#### Brief summary of issue
I switch from no delay to 4 seconds delay but the change is not applied

#### Steps to reproduce the issue
Launch the app with 0 Delay
Change it to 4 seconds
Click selection
=> the selection tool is applied immediately and the delay is ignored.

#### Extra information, such as Shutter version, display server in use (Xorg or Wayland), operating system and ideas for how to solve:
Version: 0.99.2-4
OS: Ubuntu 22.04

**Comment by vadi2** [comment]
Are you taking a new screenshot or are you doing a repeat screenshot? If it's a repeat screenshot, it is re-run with the original parameters and thus the original delay.

**Comment by djegithub** [comment]
Hi,

woaw, fast response! your tool is the best I found for Ubuntu :-)
No, it is a single screenshot. I started the app, the delay was 0, I
updated it to 4 (to have time to click on a contextual menu) and clicked on
selection.
The only way I found to have the delay applied is to click new Tooltip so
that my three screens are captured to then resize the interesting part.

Le jeu. 9 mars 2023 à 10:00, Vadim Peretokin ***@***.***> a
écrit :

**Comment by DarthGandalf** [comment]
Are you using wayland?

чт, 9 мар. 2023 г., 09:29 djegithub ***@***.***>:

**Comment by djegithub** [comment]
Google told me wayland was related to games so I would say no as I have no
idea what it was.

Le jeu. 9 mars 2023 à 10:42, Alexey Sokolov ***@***.***> a
écrit :

**Comment by djegithub** [comment]
My config
[image: image.png]

**Comment by DarthGandalf** [comment]
The image didn't show.

Can you show screenshot of main window of shutter?

чт, 9 мар. 2023 г., 09:45 djegithub ***@***.***>:

**Comment by djegithub** [comment]
Sure, I attached both then.I also noticed the timeout does not apply when
using "Selection" button but it applies if you use the "Window" one

**Comment by DarthGandalf** [comment]
You didn't attach any, please use the github ui

чт, 9 мар. 2023 г., 12:28 djegithub ***@***.***>:

**Comment by djegithub** [comment]
Attached to mail but filtered out...
![shutterWindow](https://user-images.githubusercontent.com/86766400/224047360-68fcbd9d-2c94-46c6-98c3-a218e0873af1.png)
![system](https://user-images.githubusercontent.com/86766400/224047368-f9d5aaad-0095-49a1-8fed-936f672e8333.png)

**Comment by DarthGandalf** [comment]
So yes, this is another Wayland issue. You can switch to X11 to get the full functionality of Shutter: #187 

On X11 Shutter takes the screenshot itself, but on Wayland we're currently limited to XDG portal which provides a very limited API. Depending on DE it may show some other (its own) UI where it may or may not provide ability to specify delay, but Shutter has no control over that.

That's why we already disable all buttons in toolbar except for one when running under Wayland. At least to reduce confusion, the Delay and Cursor buttons probably should be disabled as well. Until we find some other way to take screenshots on Wayland without such limitations.

**Comment by DarthGandalf** [comment]
I don't know why it says windowing system X11 in the other screenshot.
Somehow gnome thinks it's x11, but shutter thinks it's wayland?

**Comment by hsandt** [comment]
Started happening to me as well since upgrading to Ubuntu 22.04. I'm on Unity desktop (which also happens to break most of the native gnome-screenshot, so Shutter is becoming more and more vital).

**Comment by Photon89** [comment]
What is `echo $XDG_SESSION_TYPE` reporting?

**Comment by hsandt** [comment]
x11

**Comment by Photon89** [comment]
Could you please launch Shutter from command line, do the delay change, then the screenshot and then post the output from command line?

**Comment by hsandt** [comment]
So, it looks like I already had a Shutter instance running:

> INFO: There is already another instance of Shutter running!

I shut it down by clicking on the tray icon > Quit and the bug disappeared.

Then I reopened two instances, but the bug didn't reappear.

It's possible that having two instances running confuses the preferences, but since I couldn't repro, I'm not sure now.

**Comment by Photon89** [comment]
Ok, just tingle around a bit more and see if you can reproduce it systematically in some way!

**Comment by djegithub** [comment]
I come back with my info.
The wheel starts by default with "Ubuntu" choice. Other possibilities are GNOME, GNOME on Wayland and Ubuntu on Wayland.

When I run the shel command, I also get X11.

**Comment by Photon89** [comment]
Just to make sure: The capture modes "Selection", "Window" etc are disabled as is shown on your first screenshot although you are running X11?

**Comment by djegithub** [comment]
I had used shutter to test deeper and reached that situation.
By default, all buttons are enabled.
There is something really strange... Today, verything works fine. I did not upgrade shutter (still latest of the ubuntu store, i.e. 0.99.2). The only difference is that I'l running my PC from home instead of work. Could it make a difference?
I had noticed once in the past the the delay had been taken into account after an application or OS restart but today ot clearly processes everything fine.
I would say there must be something wrong as I tried to make it work more than 15 minutes and the issue was present when I created it and we did the first investigation but today, no issue...

**Comment by DarthGandalf** [comment]
Do you mean it's the same pc which you move between home and work, or
different machines with different setup?

пн, 13 мар. 2023 г., 10:34 djegithub ***@***.***>:

**Comment by djegithub** [comment]
I tried again from the office where it works as it should.
I'm sorry, I'm sure I could not get rid of the delay in "Selection" once at work but now it works fine from home and office.
I fear I can't reproduce it...

**Comment by wheelerlaw** [comment]
>Depending on DE it may show some other (its own) UI where it may or may not provide ability to specify delay, but Shutter has no control over that.

Actually Shutter does have control over this. Shutter could respect the delay setting when run under Wayland by just delaying until it makes the call to the XDG portal API. It seems that this is how ksnip handles it.

**Comment by DarthGandalf** [comment]
No. XDG portal API is free to show an arbitrary window which asks user how to do the screenshot. Delaying before that window is shown will achieve nothing. If some particular implementation of the portal shows no such prompt, and just immediately does the screenshot of the whole desktop, then delay on Shutter side would work, but we can't rely on any particular implementation, or can't even know which implementation will be called by the portal API.

**Comment by ventoux84** [comment]
On my system, "Delay is not applied". I changed from 4 to 15 and 4 is always applied.
This only occurs with menu and tooltip. Window and full desktop work fine.
```
Operating System: Arch Linux 
KDE Plasma Version: 5.27.8
KDE Frameworks Version: 5.110.0
Qt Version: 5.15.10
Kernel Version: 6.5.3-arch1-1 (64-bit)
Graphics Platform: X11
Processors: 4 × Intel® Core™ i5-6500T CPU @ 2.50GHz
Memory: 15.5 Gio of RAM
Graphics Processor: Mesa Intel® HD Graphics 530
Manufacturer: HP
Product Name: HP EliteDesk 800 G3 DM 35W
```
It's the same thing, just after boot or latter after killing and launching a new instance of Shutter.
